skimage/measure/fit.py

```diff
@@ -680,6 +849,44 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     return np.ceil(np.log(nom) / np.log(denom))
 
 
+def add_from_estimate(cls):
+    """Add ``from_estimate`` method  class using ``estimate`` method"""
+
+    if hasattr(cls, 'from_estimate'):
+        if not ismethod(cls.from_estimate):
+            raise TypeError(f'Class {cls} `from_estimate` must be a ' 'class method.')
+        return cls
+
+    if not hasattr(cls, 'estimate'):
+        raise TypeError(
+            f'Class {cls} must have `from_estimate` class method '
+            'or `estimate` method.'
+        )
```

Comment:
I wonder if we can we replace this in favor of simple `isinstance(cls, RansacModelProtocol)`. Is the reason you are doing this because of more helpful error messages?